src/client.ts

```diff
@@ -4608,7 +4641,38 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         threadId: string | null,
         eventObject: Partial<IEvent>,
         txnId?: string,
-    ): Promise<ISendEventResponse> {
+    ): Promise<ISendEventResponse>;
+    /**
+     * Sends the event as a future (MSC4140).
+     * @param eventObject - An object with the partial structure of an event, to which event_id, user_id, room_id and origin_server_ts will be added.
+     * @param futureOpts - Properties of the future to send for this event.
+     * @param txnId - Optional.
+     * @returns Promise which resolves: to an empty object `{}`
+     * @returns Rejects: with an error response.
+     */
+    private sendCompleteEvent<F extends ISendFutureRequestOpts>(
+        roomId: string,
+        threadId: string | null,
+        eventObject: Partial<IEvent>,
+        futureOpts: F,
+        txnId?: string,
+    ): Promise<ISendFutureResponse<F>>;
+    private sendCompleteEvent<F extends ISendFutureRequestOpts>(
+        roomId: string,
+        threadId: string | null,
+        eventObject: Partial<IEvent>,
+        futureOptsOrTxnId?: F | string,
+        txnIdOrVoid?: string,
+    ): Promise<ISendEventResponse | ISendFutureResponse<F>> {
```

Comment:
since this is a private member I am wondering if we can use named fields (prop obj).
```
private sendCompleteEvent(sendData: SendCompleteEventData | SendCompleteFutureEventData)
```
and then specify the the two send data interfaces.
Would also match the return Promise type.
And can be type checked easier.

But it looks like the `...Or...` naming pattern is used at other places as well. So from that point of view this should be fine.

Reply:
Yes, that naming pattern is already used elsewhere in this file & I found it to be a nice way to write overloads while getting usable type hints when calling them.